iceoryx_platform/win/source/pthread.cpp

```diff
@@ -22,6 +22,44 @@
 #include <cwchar>
 #include <vector>
 
+#if defined(__GNUC__) || defined(__GNUG__)
+int iox_pthread_setname_np(iox_pthread_t thread, const char* name)
+{
+    return pthread_setname_np(thread, name);
+}
+
+int iox_pthread_getname_np(iox_pthread_t thread, char* name, size_t len)
+{
+    return pthread_getname_np(thread, name, len);
+}
+
+int iox_pthread_create(iox_pthread_t* thread, const iox_pthread_attr_t* attr, void* (*start_routine)(void*), void* arg)
+{
+    return pthread_create(thread, attr, start_routine, arg);
+}
```

Comment:
I think I can clean this up further so that the `iox_pthread` calls are not behind a compiler switch and just add `pthread_setname_np` behind a `MSVC` switch